src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp

```diff
@@ -139,6 +139,29 @@ void ValuesBlockInputFormat::readRow(MutableColumns & columns, size_t row_num)
     ++total_rows;
 }
 
+bool ValuesBlockInputFormat::tryReadDefaultValue(IColumn & column, size_t column_idx)
+{
+    try
+    {
+        if (!checkStringByFirstCharacterAndAssertTheRestCaseInsensitive("DEFAULT", *buf))
+            return false;
+
+        skipWhitespaceIfAny(*buf);
+        assertDelimiterAfterValue(column_idx);
+    }
+    catch (const Exception & e)
+    {
+        if (!isParseError(e.code()))
+            throw;
+
+        buf->rollbackToCheckpoint();
+        return false;
+    }
+
+    column.insertDefault();
```

Comment:
This is incorrect - it should insert the default value according to the table definition.

Reply:
No, it's correct. Default expressions are evaluated and missed values are filled later. ([1](https://github.com/ClickHouse/ClickHouse/blob/eff9919833d7087810e5ded8a5919dee76fbebc1/src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp#L200-L203), [2](https://github.com/ClickHouse/ClickHouse/blob/eff9919833d7087810e5ded8a5919dee76fbebc1/src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp#L130-L132)).

Reply:
Good. Let's continue?